src/solvers/dgsem_structured/dg_3d.jl

```diff
@@ -90,6 +90,102 @@ function calc_volume_integral!(du, u,
+    for jj in (j+1):nnodes(dg)
+      u_node_jj  = get_node_vars(u, equations, dg, i, jj, k, element)
+      flux1 = volume_flux(u_node, u_node_jj, 1, equations)
+      flux2 = volume_flux(u_node, u_node_jj, 2, equations)
+      flux3 = volume_flux(u_node, u_node_jj, 3, equations)
+      # pull the contravariant vectors and compute their average
+      Ja21_node_jj, Ja22_node_jj, Ja23_node_jj = get_contravariant_vector(2, contravariant_vectors,
+                                                                          i, jj, k, element)
+      Ja21_avg = 0.5 * (Ja21_node + Ja21_node_jj)
+      Ja22_avg = 0.5 * (Ja22_node + Ja22_node_jj)
+      Ja23_avg = 0.5 * (Ja23_node + Ja23_node_jj)
+      # compute the contravariant sharp flux
+      fluxtilde2 = Ja21_avg * flux1 + Ja22_avg * flux2 + Ja23_avg * flux3
+      # save and exploit symmetry
+      set_node_vars!(ftilde2, fluxtilde2, equations, dg, j,  i, jj, k)
+      set_node_vars!(ftilde2, fluxtilde2, equations, dg, jj, i, j , k)
+    end
+
+    # contravariant fluxes in the third direction
+    for kk in (k+1):nnodes(dg)
+      u_node_kk  = get_node_vars(u, equations, dg, i, j, kk, element)
+      flux1 = volume_flux(u_node, u_node_kk, 1, equations)
+      flux2 = volume_flux(u_node, u_node_kk, 2, equations)
+      flux3 = volume_flux(u_node, u_node_kk, 3, equations)
+      # pull the contravariant vectors and compute their average
+      Ja31_node_kk, Ja32_node_kk, Ja33_node_kk = get_contravariant_vector(3, contravariant_vectors,
+                                                                          i, j, kk, element)
+      Ja31_avg = 0.5 * (Ja31_node + Ja31_node_kk)
+      Ja32_avg = 0.5 * (Ja32_node + Ja32_node_kk)
+      Ja33_avg = 0.5 * (Ja33_node + Ja33_node_kk)
+      # compute the contravariant sharp flux
+      fluxtilde3 = Ja31_avg * flux1 + Ja32_avg * flux2 + Ja33_avg * flux3
+      # save and exploit symmetry
+      set_node_vars!(ftilde3, fluxtilde3, equations, dg, k,  i, j, kk)
+      set_node_vars!(ftilde3, fluxtilde3, equations, dg, kk, i, j , k)
+    end
+  end
+
+  calcflux_twopoint_nonconservative!(ftilde1, ftilde2, ftilde3, u, element,
+                                     have_nonconservative_terms(equations),
+                                     mesh, equations, dg, cache)
+end
+
+
+function calcflux_twopoint_nonconservative!(f1, f2, f3, u, element,
+                                            nonconservative_terms::Val{true},
+                                            mesh::StructuredMesh{3},
+                                            equations, dg::DG, cache)
+  #TODO: Create a unified interface, e.g. using non-symmetric two-point (extended) volume fluxes
+  #      For now, just dispatch to an existing function for the IdealMhdEquations
+  @unpack contravariant_vectors = cache.elements
+  calcflux_twopoint_nonconservative!(f1, f2, f3, u, element, contravariant_vectors,
+                                     equations, dg, cache)
+end
+
```

Comment:
Do you use any of this stuff? I think it was removed when re-designing the nonconservative story of Trixi.jl... Merge conflicts?